index.js

```diff
@@ -479,3 +483,21 @@ function getPathFromOptions(options) {
     return null
   }
 }
+
+// offet a influxTime value by offsetTime
+function offset(influxTime, offsetTime) {
+    let tKey= influxTime.slice(-1)
+    if(isNaN(offsetTime)) { return sanitize(influxTime) }
+
+    let oVal= Number(influxTime.substring(0, influxTime.length - 1) ) + Math.abs( Number(offsetTime) )
+    return sanitize( oVal + influxDurationKeys[tKey] )
+}
+
+// calculate query end time.
+function calcEndTime(timespanValue, offsetValue) {
```

Comment:
I'd rename this `endTimeExpression` and remove the comment, as it doesn't calculate much anything.